src/Common/Core/Impl/Threading/AsyncReaderWriterLock.Queue.cs

```diff
@@ -70,7 +70,7 @@ private class Queue {
             ///          ↓     ↓            ↓        ↓
             ///    R−→W−→W−→E−→R  ―►  R−→W−→W−→E−→E−→R
             ///             ↑                     ↑
-            ///   erLock.Tail               erlTail
+            ///       erlTail               erlTail
             /// </remarks>
             public void AddExclusiveReader(ExclusiveReaderLockSource item, out bool isAddedAfterWriterOrExclusiveReader) {
```

Comment:
The lock may benefit from detailed documentation. What is exactly the sync pattern? Single writer/multiple readers? What is the example of exclusive reader? Can it simply be a writer that actually doesn't write anything?